app/models/user.rb

```diff
@@ -143,4 +143,13 @@ def sync_app_installation_access
     removed_permissions = app_installation_permissions.reject{|ep| app_installation_ids.include?(ep.app_installation_id) }
     removed_permissions.each(&:destroy)
   end
+
+  def can_comment?(subject)
+    return subject.commentable? && subject.repository.commentable? && #is the subject/repo commentable? (checkes for write issues)
+      personal_access_token_enabled? || #do we have a personal access token?
+      Octobox.fetch_subject? || # are we running with repo scope?      
+      subject.repository.app_installation_id || #there is an app installation on the repo (we already checked for write permission)
```

Comment:
You say `we already checked for write permission` but I can't see where this check is happening?

Reply:
> You say `we already checked for write permission` but I can't see where this check is happening?

Ok, it looks like that's happening in `subject.repository.commentable?` but the `||` are shortcutting that check after the first one, would be good to either add brackets around the groups of conditions or break this out into separate checks so it's easier to read.